### PR TITLE
Fixed more bugs

### PR DESCRIPTION
**Pull request by ank266**
Fixed the bug where changing date format would say data is empty and where not selecting a file would say file is invalid and fixed bug where selecting a valid file than selecting an invalid file than selecting a date format graphs old valid file

### CODE REVIEW

File: src/components/ParserComponent.tsx [hunk 1]
```diff
@@ -92,11 +93,15 @@ export default class ParserComponent extends React.Component<ParserInterface,
                   });
                   const mockDateFile: File = new File(
                       [this.state.fileData],
-                      'mockFile.csv',
+                      String(this.state.upFile),
                       {type: this.props.fileType.mimeName},
                   );
                   const fileEvent = {target: {files: [mockDateFile]}};
-                  this.parse(fileEvent);
+                  // eslint-disable-next-line max-len
```
Comment (braunsonm) [inline src/components/ParserComponent.tsx]: Split this line if it's too long. No need to ignore a lint rule

File: src/components/ParserInterface.ts [hunk 2]
```diff
@@ -38,4 +38,5 @@ export interface ParserState {
     showTimeline: boolean;
     formatString: string;
     fileData: any;
+    upFile: any;
```
Comment (braunsonm) [inline src/components/ParserInterface.ts]: Can you add documentation about what `upFile` is?

File: src/components/ParserComponent.tsx [hunk 3]
```diff
@@ -387,23 +423,25 @@ export default class ParserComponent extends React.Component<ParserInterface,
...
           showTimeline: false,
         };
       });
 
       const temp: File = fileEvent.target.files[0];
       if (this.isValid(temp)) {
         await this.parseCsv(fileEvent);
+        // only show timeline if there is data
+        assert.notStrictEqual(this.state.data, [],
+            'parse(): this.state.data is empty ' +
```
Comment (camilleion) [inline src/components/ParserComponent.tsx]: Whoo good assertion!

File: src/__tests__/ParserComponent.test.tsx [hunk 4]
```diff
@@ -798,7 +873,7 @@ describe('<ParserComponent /> Unit Tests', () => {
     it('timeline is set to true', async () => {
       // Call the parse method with the fake event
       await comp.instance().parse(event);
-      expect(comp.state('showTimeline')).toBeTruthy();
+      expect(comp.state('showTimeline')).toBeFalsy();
```
Comment (braunsonm) [inline src/__tests__/ParserComponent.test.tsx]: Why'd you change this? The it block doesn't make sense anymore and shouldn't it be Truthy?

File: src/components/ParserComponent.tsx [hunk 5]
```diff
@@ -121,6 +114,40 @@ export default class ParserComponent extends React.Component<ParserInterface,
...
     }
 
+    /**
+     * Purpose: create a mock file event of the actual file for
+     * parse when it is recalled everytime the date format is changed
+     * @preconditions: a file with valid file data and
+     * a valid file name and a valid file type
+     * @return {any}: a mock file event of the actual file
+     */
+    createNewMockFile(): any {
```
Comment (braunsonm) [inline src/components/ParserComponent.tsx]: Why do we need a mock file? Why can't it just be undefined if it's not set?

File: src/components/ParserComponent.tsx [hunk 6]
```diff
@@ -121,6 +114,40 @@ export default class ParserComponent extends React.Component<ParserInterface,
...
+   * Purpose: check if file is .csv when date format
+     * is changed and call parse if it is
+     * @preconditions: the current file should have a valid name (.csv)
+     * @postconditions: parse is called if file is .csv
+     * @return{boolean}: returns true if it works otherwise returns false
+   */
+    checkifCsvandcallParse(): boolean {
+      const nameOfFile = this.state.fileName;
+      const typeOfFile = nameOfFile.substr(nameOfFile.length - 4);
+      if (typeOfFile === '.csv') {
```
Comment (braunsonm) [inline src/components/ParserComponent.tsx]: Didn't we already do this in other places? And what about a .pdf that has its file ending changed to .csv?

File: src/components/ParserComponent.tsx (hunk 5 above)
Comment (ank266) [reply src/components/ParserComponent.tsx]: we need mock file event to put in parse as it needs a file event argument

**Pull request merged**